ethcore/engines/authority-round/src/lib.rs

```diff
@@ -830,30 +836,21 @@ impl AuthorityRound {
 		let initial_step = our_params.start_step.unwrap_or(0);
 
 		let mut durations = Vec::new();
-		let mut prev_step = 0u64;
-		let mut prev_time = 0u64;
-		let mut prev_dur = our_params.step_durations[&0];
-		durations.push(StepDurationInfo {
-			transition_step: prev_step,
-			transition_timestamp: prev_time,
-			step_duration: prev_dur
-		});
-		for (time, dur) in our_params.step_durations.iter().skip(1) {
-			let (step, time) = next_step_time_duration(
-				StepDurationInfo{
-					transition_step: prev_step,
-					transition_timestamp: prev_time,
-					step_duration: prev_dur,
-				}, *time)
-				.ok_or(BlockError::TimestampOverflow)?;
-			durations.push(StepDurationInfo {
-				transition_step: step,
-				transition_timestamp: time,
-				step_duration: *dur
-			});
-			prev_step = step;
-			prev_time = time;
-			prev_dur = *dur;
+		{
+			let mut dur_info = StepDurationInfo {
+				transition_step: 0u64,
+				transition_timestamp: 0u64,
+				step_duration: our_params.step_durations[&0],
+			};
+			durations.push(dur_info);
+			for (time, dur) in our_params.step_durations.iter().skip(1) {
+				let (step, time) = next_step_time_duration(dur_info, *time)
+					.ok_or(BlockError::TimestampOverflow)?;
+				dur_info.transition_step = step;
+				dur_info.transition_timestamp = time;
+				dur_info.step_duration = *dur;
+				durations.push(dur_info);
```

Comment:
This works because `StepDurationInfo` is `Copy`?

Reply:
Yes; I made it copy because it's just three `u64`s anyway.

Reply:
Cool, makes sense. I wonder if `durations.push(dur_info.clone())` would compile down to a no-op (and if it does, would that make things clearer or be more confusing? cc @ordian ?).

Reply:
I think it would be equivalent, and it would trigger another [`clone_on_copy`](https://rust-lang.github.io/rust-clippy/master/#clone_on_copy) clippy warning.